routers/repo/view.go

```diff
@@ -54,18 +54,38 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		return
 	}
 
-	var readmeFile *git.Blob
+	var readmes []struct {
+		*git.Blob
+		tp string
+	}
 	for _, entry := range entries {
-		if entry.IsDir() || !markup.IsReadmeFile(entry.Name()) {
+		if entry.IsDir() {
+			continue
+		}
+
+		tp, ok := markup.ReadmeFileType(entry.Name())
+		if !ok {
 			continue
 		}
 
-		// TODO: collect all possible README files and show with priority.
-		readmeFile = entry.Blob()
-		break
+		readmes = append(readmes, struct {
+			*git.Blob
+			tp string
+		}{
+			entry.Blob(),
+			tp,
+		})
 	}
 
-	if readmeFile != nil {
+	if len(readmes) > 0 {
+		var readmeFile *git.Blob
+		for _, readme := range readmes {
```

Comment:
Why re-loop if we stop at first ok. This could be done directly in the first loop at https://github.com/lunny/gitea/blob/d0ddef4e095a8f4e317a6163cb2e955fea254779/routers/repo/view.go#L67 for example ?

This just add an additional test if the file is a good filetype so no need to save them in an array ?